driver/miopen_ReductionHost.hpp

```diff
@@ -191,15 +190,29 @@ class miopenReductionHost
     template <typename compType>
     void RunImpl(Tgpu alpha, const Tgpu* in_data, Tgpu beta, Tref* out_data, int* indices)
     {
-        using namespace reduce;
+        bool need_indices =
+            (indicesOpt == MIOPEN_REDUCE_TENSOR_FLATTENED_INDICES) &&
+            (reduceOp == MIOPEN_REDUCE_TENSOR_MIN || reduceOp == MIOPEN_REDUCE_TENSOR_MAX);
+
+        if(need_indices)
+            RunImpl_with_indices<compType>(alpha, in_data, beta, out_data, indices);
+        else
+            RunImpl_no_indices<compType>(alpha, in_data, beta, out_data);
```

Comment:
Now there are two 100-line functions that differ in ~10 lines only. Unfortunately, this is not acceptable. If this is a clang-tidy requirement, then let's git rid od clang-tidy.